src/jit/jiteh.cpp

```diff
@@ -1075,7 +1075,7 @@ void* Compiler::ehEmitCookie(BasicBlock* block)
 
     void* cookie;
 
-#if FEATURE_EH_FUNCLETS && defined(_TARGET_ARM_)
+#if FEATURE_EH_FUNCLETS && defined(_TARGET_ARM_) && defined(LEGACY_BACKEND)
     if (block->bbFlags & BBF_FINALLY_TARGET)
```

Comment:
Yes, that was the place I had doubts about. Now, after I've looked at it more carefully, I realized that I've missed that nop insertion workaround for ARM32 unwinder at [codegenlegacy.cpp#L12866-L12894](https://github.com/dotnet/coreclr/blob/abbb8f685929c7aeaa087dae46fedc1bc2af4b17/src/jit/codegenlegacy.cpp#L12866-L12894).
I've moved this piece into a separate function to prevent duplication.